src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Orleans.Providers.Streams.Common;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    public class EventHubQueueCacheFactory : IEventHubQueueCacheFactory
+    {
+        private readonly EventHubStreamProviderSettings _providerSettings;
+        private readonly SerializationManager _serializationManager;
+        private readonly FixedSizeObjectPool<FixedSizeBuffer> _bufferPool;
+        private readonly TimePurgePredicate _timePurge;
+
+        public EventHubQueueCacheFactory(EventHubStreamProviderSettings providerSettings, SerializationManager serializationManager)
+        {
+            _providerSettings = providerSettings;
+            _serializationManager = serializationManager;
+
+            _bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(_providerSettings.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
+            _timePurge = new TimePurgePredicate(_providerSettings.DataMinTimeInCache, _providerSettings.DataMaxAgeInCache);
+        }
+
+        public IEventHubQueueCache CreateCache(string partition, IStreamQueueCheckpointer<string> checkpointer, Logger cacheLogger)
+        {
+            var cache = CreateCache(checkpointer, cacheLogger, _bufferPool, _timePurge, _serializationManager);
+            if (_providerSettings.AveragingCachePressureMonitorFlowControlThreshold.HasValue)
```

Comment:
suggest "AddPressurceMonitors" virtual function to specialize function, for those that want to customize the pressure monitoring.